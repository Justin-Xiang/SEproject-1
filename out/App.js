import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import './App.css';
import { MainContent } from './components/MainContent';
import { SideMenu } from './components/SideMenu';
function App() {
    return (_jsxs(_Fragment, { children: [_jsx(SideMenu, {}, void 0), _jsx(MainContent, {}, void 0)] }, void 0));
}
export default App;
