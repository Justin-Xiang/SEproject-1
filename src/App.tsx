import React from 'react';
import './App.css';
import { MainContent } from './components/MainContent';
import { MenuItem, MyMenu, SubMenu } from './components/SideMenu';
import { user } from './utils/store';
function App() {
	return (
		<>
			<MyMenu>
				<MenuItem key={1}>首页</MenuItem>
				<MenuItem key={2}>首页</MenuItem>
				<SubMenu key={3} title="首页">
					<MenuItem key={4}>首页</MenuItem>
				</SubMenu>
			</MyMenu>
			<MainContent user={user} />
		</>
	);
}

export default App;
