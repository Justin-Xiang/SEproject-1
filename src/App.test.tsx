/* eslint-disable no-undef */
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import App from './App';
React.useLayoutEffect = React.useEffect;
Enzyme.configure({ adapter: new Adapter() });
describe('Button 组件测试', () => {
	it('渲染正常', () => {
		// enzyme的shallow方法用于渲染组件
		const componentWrapper = shallow(<App />);
		expect(componentWrapper.html()).toMatchSnapshot();
	});
});
