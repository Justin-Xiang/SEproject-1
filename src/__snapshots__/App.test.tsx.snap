// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 组件测试 渲染正常 1`] = `"<div class=\\"contain-sideMenu\\"><div><div><img alt=\\"\\" src=\\"../image/IMG.jpg\\"/></div></div><div class=\\"contain-menu\\"><ul class=\\"menu\\"><li class=\\"menu-item\\"><a href=\\"#/\\"><span class=\\"menu-item-title\\">首页</span></a></li><li class=\\"menu-submenu-item\\"><div class=\\"menu-submenu-item-title\\"><span class=\\"menu-item-title\\">题库</span><i class=\\"menu-submenu-item-arrow\\"></i></div><ul class=\\"menu-submenu\\" style=\\"max-height:0;overflow:hidden\\"><li class=\\"menu-item menu-item-child\\"><a href=\\"#/question\\"><span class=\\"menu-item-title\\">一年级</span></a></li><li class=\\"menu-item menu-item-child\\"><a href=\\"#/question\\"><span class=\\"menu-item-title\\">二年级</span></a></li><li class=\\"menu-item menu-item-child\\"><a href=\\"#/question\\"><span class=\\"menu-item-title\\">三年级</span></a></li><li class=\\"menu-item menu-item-child\\"><a href=\\"#/question\\"><span class=\\"menu-item-title\\">四年级</span></a></li><li class=\\"menu-item menu-item-child\\"><a href=\\"#/question\\"><span class=\\"menu-item-title\\">五年级</span></a></li><li class=\\"menu-item menu-item-child\\"><a href=\\"#/question\\"><span class=\\"menu-item-title\\">六年级</span></a></li></ul></li><li class=\\"menu-item\\"><a href=\\"#/wrongSet\\"><span class=\\"menu-item-title\\">错题集</span></a></li><li class=\\"menu-item\\"><a href=\\"#/history\\"><span class=\\"menu-item-title\\">历史记录</span></a></li></ul></div></div><div class=\\"mainContent\\"><div class=\\"box-3d\\"><div class=\\"label-contain\\"><span class=\\"log-labels\\" id=\\"log-label\\">Log in</span><span class=\\"log-labels\\" id=\\"sign-label\\">Sign up</span></div><input type=\\"checkbox\\" class=\\"checkbox\\" id=\\"log\\"/><label class=\\"checkbox-for\\" for=\\"log\\"><span class=\\"checkbox-for-icon\\"><i class=\\"uil uil-arrow-up-left\\"></i></span></label><div class=\\"wrap-3d\\"><div class=\\"wrapper-3d\\"><div class=\\"wrap-3d-front\\"><div class=\\"login out\\"><div class=\\"login\\"><form class=\\"ant-form ant-form-horizontal\\"><div class=\\"login-container\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"user\\" class=\\"anticon anticon-user\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"user\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z\\"></path></svg></span></span><input type=\\"text\\" maxLength=\\"100\\" placeholder=\\"用户名\\" value=\\"\\" id=\\"username\\" class=\\"ant-input ant-input-lg\\"/></span></div></div></div></div><p style=\\"margin:-7px 0;display:block\\"> </p><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"lock\\" class=\\"anticon anticon-lock\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"lock\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z\\"></path></svg></span></span><input type=\\"password\\" maxLength=\\"100\\" placeholder=\\"密码\\" action=\\"click\\" value=\\"\\" id=\\"password\\" class=\\"ant-input ant-input-lg\\"/><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"eye-invisible\\" tabindex=\\"-1\\" class=\\"anticon anticon-eye-invisible ant-input-password-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"eye-invisible\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z\\"></path><path d=\\"M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z\\"></path></svg></span></span></span></div></div></div></div></div><div class=\\"submit\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" id=\\"remember\\" class=\\"ant-checkbox-input\\"/><span class=\\"ant-checkbox-inner\\"></span></span><span>Remember me</span></label></div></div></div></div><div class=\\"ant-row ant-form-item\\" style=\\"justify-content:center\\"><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div style=\\"display:flex;justify-content:center\\"><button style=\\"width:80px;height:40px\\" type=\\"submit\\" class=\\"ant-btn ant-btn-primary\\"><span>登 录</span></button></div></div></div></div></div></div></form></div></div></div><div class=\\"wrap-3d-back\\"><div class=\\"login out\\"><div class=\\"sign\\"><form class=\\"ant-form ant-form-horizontal\\"><div class=\\"sign-container\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"user\\" class=\\"anticon anticon-user\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"user\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z\\"></path></svg></span></span><input type=\\"text\\" maxLength=\\"100\\" placeholder=\\"邮箱\\" value=\\"\\" id=\\"email\\" class=\\"ant-input ant-input-lg\\"/></span></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"user\\" class=\\"anticon anticon-user\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"user\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z\\"></path></svg></span></span><input type=\\"text\\" maxLength=\\"100\\" placeholder=\\"用户名\\" value=\\"\\" id=\\"username\\" class=\\"ant-input ant-input-lg\\"/></span></div></div></div></div><p style=\\"margin:-7px 0;display:block\\"> </p><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"lock\\" class=\\"anticon anticon-lock\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"lock\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z\\"></path></svg></span></span><input type=\\"password\\" maxLength=\\"100\\" placeholder=\\"密码\\" action=\\"click\\" value=\\"\\" id=\\"password\\" class=\\"ant-input ant-input-lg\\"/><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"eye-invisible\\" tabindex=\\"-1\\" class=\\"anticon anticon-eye-invisible ant-input-password-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"eye-invisible\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z\\"></path><path d=\\"M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z\\"></path></svg></span></span></span></div></div></div></div></div><div class=\\"submit\\"><div class=\\"ant-row ant-form-item\\" style=\\"justify-content:center\\"><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div style=\\"display:flex;justify-content:center\\"><button style=\\"width:80px;height:40px\\" type=\\"submit\\" class=\\"ant-btn ant-btn-primary\\"><span>注 册</span></button></div></div></div></div></div></div></form></div></div></div></div></div></div></div>"`;
